syntax = "proto3";

package fixbug;

// Enable generation of generic C++ services
option cc_generic_services = true;

message ResultCode {
    int32 errcode = 1;
    bytes errmsg = 2;
}

// 登录
message LoginRequest {
    bytes username = 1;
    bytes password = 2;
}

message LoginResponse {
    ResultCode result = 1;
    bool success = 2;
}

// 注册
message RegisterRequest {
    uint32 id = 1;
    bytes username = 2;
    bytes password = 3;
}

message RegisterResponse {
    ResultCode result = 1;
    bool success = 2;
}

service UserServiceRPC {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}